diff --git a/activity.py b/activity.py
index f84da99..c96e1e2 100644
--- a/activity.py
+++ b/activity.py
@@ -22,16 +22,16 @@ from gi.repository import Gtk
 import pygame
 import sugargame
 import sugargame.canvas
-from sugar3.activity import activity
 from sugar3.graphics.toolbarbox import ToolbarBox
-from sugar3.activity.widgets import ActivityToolbarButton
 from sugar3.activity.widgets import StopButton
 import cells
 
+from sugarapp.widgets import SugarCompatibleActivity
+from sugarapp.widgets import ExtendedActivityToolbarButton
 
-class Activity(activity.Activity):
+class Activity(SugarCompatibleActivity):
     def __init__(self, handle):
-        activity.Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
 
         self.max_participants = 1
         self.game = cells.Game()
@@ -49,7 +49,7 @@ class Activity(activity.Activity):
         self.set_toolbar_box(toolbar_box)
         toolbar_box.show()
 
-        activity_button = ActivityToolbarButton(self)
+        activity_button = ExtendedActivityToolbarButton(self)
         toolbar_box.toolbar.insert(activity_button, -1)
         activity_button.show()
 
